vtrace/platforms/linux.py

```diff
@@ -307,18 +307,27 @@ def platformExec(self, cmdline):
         self.execing = True
         cmdlist = e_cli.splitargs(cmdline)
         os.stat(cmdlist[0])
-        pid = os.fork()
-
+        
+        pid = os.fork() 
+        
+        # Child will be blocked after execv because of trace me!
         if pid == 0:
             v_posix.ptrace(v_posix.PT_TRACE_ME, 0, 0, 0)
-            # Make sure our parent gets some cycles
-            time.sleep(0.1)
             os.execv(cmdlist[0], cmdlist)
             sys.exit(-1)
 
-        if v_posix.ptrace(PT_ATTACH, pid, 0, 0) != 0:
-            raise Exception("PT_ATTACH failed! linux platformExec")
-
+        # Parent will wait for the child
+        for i in xrange(0, 33):
```

Comment:
Under the assumption that as long as we block the child will get cpu time eventually and we are guaranteed to receive a signal from the kernel eventually, we should try to removing WNOHANG and no longer looping. The child will generate his signal to the parent (and block ) at TRACEME and we should be set to go?  Thoughts?